httpclient/src/main/java/org/apache/http/conn/ssl/SSLConnectionSocketFactory.java

```diff
@@ -183,6 +187,15 @@ public static SSLConnectionSocketFactory getSocketFactory() throws SSLInitializa
         return new SSLConnectionSocketFactory(SSLContexts.createDefault(), getDefaultHostnameVerifier());
     }
 
+    private static boolean isWeakCipherSuite(final String cipherSuite) {
+        for (final String weakAlgorithm : WEAK_ALGORITHMS) {
+            if (cipherSuite.contains(weakAlgorithm)) {
```

Comment:
@artem-smotrakov Could this be replaced with regexp matching?

Reply:
@ok2c I thought a list of weak algorithms is more straightforward. Regex would allow to build more complex constraints but not sure if it's necessary. TLS cipher suites have a pretty simple format where a particular crypto algorithm may be used only in particular part of a cipher suite.

Reply:
@artem-smotrakov Matching things by a simple occurrence of a expression at any position in the algorithm name looks really shoddy to me. I have no idea if this is a common practice but it looks sloppy.

Reply:
@ok2c I think strings from `WEAK_ALGORITHMS` can occur only in specific parts of a cipher suite. I am not sure about common practices either, but in this particular case, I think that having a list of weak algorithms looks clearer and simpler to understand rather than a regex. Anyway, if you think it has to be updated to use a regex, I'll do that. Please let me know.

Reply:
> WEAK_ALGORITHMS can occur only in specific parts of a cipher suite

@artem-smotrakov This is precisely what bothers me. The weak algorithms can occur at specific parts but the propose change-set simply tests for occurrence of those expressions at any arbitrary position in the cipher name. This looks sloppy.

Reply:
@ok2c The weak algorithms can't occur at arbitrary position in the cipher suites defined in TLS specifications. That's why I think it's enough to search for specific weak algorithms in a cipher suite. Again, no problem with updating it to use regex, I just don't think it's worth it.

Reply:
Just noticed one more thing. It may be better to use convert the string to upper-case before search or matching. The JSSE spec currently mentions only _anon_ but not sure if _ANON_ may not appear.

Reply:
@artem-smotrakov Yes, this must be fixed.